libcxx/utils/ci/buildkite-pipeline.yml

```diff
@@ -28,916 +28,509 @@ env:
+
+  - label: Clang-cl (no vcruntime exceptions)
+    command: bash libcxx/utils/ci/run-buildbot clang-cl-no-vcruntime
+    <<: *common
+    <<: *windows_agent
+
+  - label: Clang-cl (Debug mode)
+    command: bash libcxx/utils/ci/run-buildbot clang-cl-debug
+    <<: *windows_agent
+    <<: *common
+
+  - label: Clang-cl (Static CRT)
+    command: bash libcxx/utils/ci/run-buildbot clang-cl-static-crt
+    <<: *windows_agent
+    <<: *common
+
+  - label: MinGW (DLL, x86_64)
+    command: bash libcxx/utils/ci/run-buildbot mingw-dll
+    <<: *windows_agent
+    <<: *common
+
+  - label: MinGW (Static, x86_64)
+    command: bash libcxx/utils/ci/run-buildbot mingw-static
+    <<: *windows_agent
+    <<: *common
+
+  - label: MinGW (DLL, i686)
+    command: bash libcxx/utils/ci/run-buildbot mingw-dll-i686
+    <<: *windows_agent
+    <<: *common
+
+- group: ':mac: Apple'
+  steps:
+  - label: MacOS x86_64
+    command: libcxx/utils/ci/run-buildbot generic-cxx23
+    <<: *mac_agent
+    <<: *common
+
+  - label: MacOS arm64
+    command: libcxx/utils/ci/run-buildbot generic-cxx23
     agents:
-      queue: "libcxx-builders"
-      os: "linux"
-    retry:
-      automatic:
-        - exit_status: -1  # Agent was lost
-          limit: 2
-    timeout_in_minutes: 120
+      queue: libcxx-builders
+      os: macos
+      arch: arm64
+    <<: *common
 
-  - label: "Debug mode"
-    command: "libcxx/utils/ci/run-buildbot generic-debug-mode"
-    artifact_paths:
-      - "**/test-results.xml"
-      - "**/*.abilist"
-    env:
-        CC: "clang-${LLVM_HEAD_VERSION}"
-        CXX: "clang++-${LLVM_HEAD_VERSION}"
-        ENABLE_CLANG_TIDY: "On"
+  - label: MacOS with Modules
+    command: libcxx/utils/ci/run-buildbot generic-modules
     agents:
-      queue: "libcxx-builders"
-      os: "linux"
-    retry:
-      automatic:
-        - exit_status: -1  # Agent was lost
-          limit: 2
-    timeout_in_minutes: 120
+      queue: libcxx-builders
+      os: macos
+    <<: *common
 
-  - label: "With LLVM's libunwind"
-    command: "libcxx/utils/ci/run-buildbot generic-with_llvm_unwinder"
-    artifact_paths:
-      - "**/test-results.xml"
-      - "**/*.abilist"
-    env:
-        CC: "clang-${LLVM_HEAD_VERSION}"
-        CXX: "clang++-${LLVM_HEAD_VERSION}"
-        ENABLE_CLANG_TIDY: "On"
+  - label: Apple system
+    command: libcxx/utils/ci/run-buildbot apple-system
     agents:
-      queue: "libcxx-builders"
-      os: "linux"
-    retry:
-      automatic:
-        - exit_status: -1  # Agent was lost
-          limit: 2
-    timeout_in_minutes: 120
-
-  - label: "Modular build with Local Submodule Visibility"
-    command: "libcxx/utils/ci/run-buildbot generic-modules-lsv"
-    artifact_paths:
-      - "**/test-results.xml"
-      - "**/*.abilist"
+      queue: libcxx-builders
+      os: macos
+      arch: arm64
+    <<: *common
+
+  - label: Apple back-deployment macosx10.13
+    command: libcxx/utils/ci/run-buildbot apple-system-backdeployment-10.13
```

Comment:
For these back-deployment builders, it doesn't make sense to use only `mac_agent`. We need to pin them to `x86_64` because of the comment that used to be there `# We need to use x86_64 for back-deployment CI on this target since macOS didn't support arm64 back then.`.

Otherwise, if we remove the workaround in `mac_agent`, we would break those back-deployment jobs.

Reply:
So we would need `mac_agent_x86` and `mac_agent_arm64`?